website/docs/d/namespace.html.markdown

```diff
@@ -0,0 +1,55 @@
+---
+layout: "kubernetes"
+page_title: "Kubernetes: kubernetes_namespace"
+sidebar_current: "docs-kubernetes-data-source-namespace"
+description: |-
+  Queries attributes of a Namespace within the cluster.
+---
+
+# kubernetes_namespace
+
+This data source provides a mechanism to query attributes of any specific namespace within a Kubernetes cluster.
+In Kubernetes, namespaces provide a scope for names and are intended as a way to divide cluster resources between multiple users.
+
+## Example Usage
+
+```hcl
+data "kubernetes_namespace" "example" {
+  metadata {
+    name = "kube-system"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `metadata` - (Required) Standard object metadata. For more info see [Kubernetes reference](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata)
+
+## Nested Blocks
+
+### `metadata`
+
+#### Arguments
+
+* `name` - (Required) Name of the namespace, must be unique. For more info see [Kubernetes reference](http://kubernetes.io/docs/user-guide/identifiers#names)
+
+#### Attributes
+
+* `annotations` - (Optional) An unstructured key value map stored with the namespace that may be used to store arbitrary metadata.
```

Comment:
Probably not needed to specify `(Optional)` as it is an export only attribute, is it?

Reply:
true!